app/controllers/users.js

```diff
@@ -1,26 +1,60 @@
 const util = require('util');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
 
 const User = require('../models').users;
 const logger = require('../logger');
 const error = require('../errors');
-const { hashPassword } = require('../services/users');
+const { hashPassword, findByEmail } = require('../services/users');
 
-const { userCreationError } = error;
+const { userCreationError, userLoginError } = error;
 
-exports.postUser = (req, res, next) => {
+exports.signUpUser = (req, res, next) => {
   const { first_name: firstName, last_name: lastName, email, password } = req.body;
-  return hashPassword(password)
-    .then(() =>
-      User.create({ firstName, lastName, email, password })
-        .then(user => {
-          logger.info(`Usuario creado para: ${firstName} ${lastName}`);
-          return user;
+  return findByEmail(email)
+    .then(foundUser => {
+      if (foundUser) {
+        throw userCreationError(`User for ${email} already exists!`);
+      } else {
+        return hashPassword(password)
+          .then(hashedPassword =>
+            User.create({ firstName, lastName, email, password: hashedPassword })
```

Comment:
Sorry for not mark in the previous PR, but you must create a service to create user!

Reply:
no problem :ok_hand: